src/librustc_mir/borrow_check/nll/renumber.rs

```diff
@@ -62,6 +62,23 @@ impl<'a, 'tcx> MutVisitor<'tcx> for NLLVisitor<'a, 'tcx> {
         debug!("visit_ty: ty={:?}", ty);
     }
 
+    fn visit_place(&mut self,
+        place: &mut Place<'tcx>,
+        context: PlaceContext,
+        location: Location) {
+        self.visit_place_base(&mut place.base, context, location);
+
+        let new_projection: Vec<_> = place.projection.iter().map(|elem|
```

Comment:
This is a very common pattern. Create a helper function that does all the repeated logic and takes a closure that gets a `&PlaceElem` as the argument and returns a `PlaceElem`.